tests/calculator-homework.spec.js

```diff
@@ -0,0 +1,116 @@
+const { test, expect } = require('@playwright/test');
+const {calcStartPage} = require('../pages/calcStartPage');
+
+test.describe('Calculator test suite', () => {
+  let page;
+  test.beforeAll(async ({ browser }) => {
+    page = await browser.newPage();
+      startPage = new calcStartPage(page);
+      
+  });
+  test.beforeEach(async () => {
+    await startPage.goto();
+  });
+
+const buildOptions = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+
+    buildOptions.forEach(buildOption => {
+    test(`Check if Build Option ${buildOption} has a Second Number field`, async () => {
+        await page.selectOption('select[name="selectBuild"]', (buildOption));
+        const isFieldVisible = await page.isVisible('#number2Field');
+        expect(isFieldVisible).toBe(true)
+    });
+  }); 
+
+
+  buildOptions.forEach(buildOption => {
+    test(`Check if Add function provides the correct answers with Build Option ${buildOption}`, async () => {
+        await page.selectOption('select[name="selectBuild"]', (buildOption));
+        await page.click('input[name="number1"]');
+        await page.fill('input[name="number1"]', '1');
+        await page.click('input[name="number2"]');
+        await page.fill('input[name="number2"]', '1');
+        await page.click('input:has-text("Calculate")');
+        const value = await page.$eval('#numberAnswerField', (el) => el.value);
+        console.log(value);
+        expect(value).toEqual('2')
+    });
+  }); 
+
+
+  buildOptions.forEach(buildOption => {
+    test(`Check that Build Option ${buildOption} returns correct error when dividing by 0`, async () => {
+        await page.selectOption('select[name="selectBuild"]', (buildOption));
+        await page.click('input[name="number1"]');
+        await page.fill('input[name="number1"]', '1');
+        await page.click('input[name="number2"]');
+        await page.fill('input[name="number2"]', '0');
+        await page.selectOption('select[name="selectOperation"]', '3');
+        await page.click('input:has-text("Calculate")');
+        const isErrorVisible = await page.isVisible('#errorMsgField');
+        expect(isErrorVisible).toBe(true);
+        const errorTextContent = await page.textContent('#errorMsgField');
+        expect(errorTextContent).toContain('Divide by zero error!'); 
+    });
+  }); 
+
+
+  buildOptions.forEach(buildOption => {
+    test(`Checks that Build Option ${buildOption} shows the correct error for non-numerical values`, async () => {
+        await page.selectOption('select[name="selectBuild"]', (buildOption));
```

Comment:
I see many places the same code lines are used - select build, fill data, click calculate and so on. These should be extracted into separate function so no code repetition and tests are much cleaner.